### PR TITLE
Merge workflow to single rspec.yml with multiple ruby versions to test

### PR DESCRIPTION
**Pull request by CAMOBAP**

### CODE REVIEW

**Comment by CAMOBAP** [comment]
For metanorma, we support ruby >= 2.4. @ronaldtse should we support ruby 2.4 for fontist too?

**Comment by ronaldtse** [comment]
@CAMOBAP if possible we'd like to support 2.4. Is it a difficult fix?

**Comment by alexeymorozov** [comment]
Looks awesome! 👍

File: .github/workflows/rspec.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          restore-keys: bundle-${{ matrix.os }}-${{ matrix.ruby }}
+
+      - if: matrix.os == 'macos-latest'
+        run: brew install lcdf-typetools
+
+      - run: bundle config set path 'vendor/bundle'
+
+      - run: bundle install --jobs 4 --retry 3
+
+      - run: bundle exec rspec --tag ~dev
```
Comment (alexeymorozov) [inline .github/workflows/rspec.yml:50]: We need somehow to handle the MacOS case: its the only OS that runs `dev` specs:

```
      - name: Run tests
        run: TEST_ENV=CI bin/rspec
```

**Review by abunashir — APPROVED** [review]
Looks good to me, regarding ruby 2.4 support: I think we might have to replace those `Hash#slice` with compatible methods as those were introduced in ruby 2.5, but then again I would leave that up to you guys  if that's something we want to continue to support in the long run - https://github.com/fontist/fontist/search?q=slice

**Comment by ronaldtse** [comment]
Thanks guys! @CAMOBAP can you help fix those two Hound issues and we can merge this?

File: .github/workflows/rspec.yml (hunk 1 above)
Comment (CAMOBAP) [reply .github/workflows/rspec.yml]: fixes

**Comment by ronaldtse** [comment]
Thanks @CAMOBAP !

**Pull request merged**